lib/ansible/modules/windows/win_firewall_rule.py

```diff
@@ -122,4 +122,14 @@
     profiles: private
     state: present
     enabled: yes
+    
+- name: Firewall rule to allow ICMP v4 echo (ping) 
+  win_firewall_rule:
+    name: ICMP Allow incoming V4 echo request
+    enabled: yes
+    state: present
+    profiles: private
+    action: allow
+    direction: in
+    protocol: "icmpv4:8,any"
```

Comment:
Is there any documentation around this protocol?

Reply:
_Is there not already a builtin rule that we can just enable?_
I didn't find one, and none of the examples on the internet worked for me, e.g. 
- name: Allow ICMP
  win_firewall_rule:
    name: File and Printer Sharing (Echo Request - ICMPv4-In)
    action: allow
    direction: in
    enabled: yes
from https://github.com/ansible/ansible/issues/32478

Environment:
ansible 2.7.9 on OSX  via homebrew
Windows 7 SP1, Windows 8.1, Windows 10 update 1809

_Is there any documentation around this protocol?_
I didn't find any ansible documentation around this, but there are many examples on the internet of using the "netsh advfirewall" command to accomplish this:
`
netsh advfirewall firewall add rule name="ICMP Allow incoming V4 echo request" protocol=icmpv4:8,any dir=in action=allow
`
https://www.howtogeek.com/howto/windows-vista/allow-pings-icmp-echo-request-through-your-windows-vista-firewall/

I found this switch statement in ansible/lib/ansible/modules/windows/win_firewall_rule.ps1 that suggests this is possible, but I don't know how ansible is invoking that script...
    switch -wildcard ($protocol) {
        "tcp" { return [System.Net.Sockets.ProtocolType]::Tcp -as [int] }
        "udp" { return [System.Net.Sockets.ProtocolType]::Udp -as [int] }
        "icmpv4*" { return [System.Net.Sockets.ProtocolType]::Icmp -as [int] }
        "icmpv6*" { return [System.Net.Sockets.ProtocolType]::IcmpV6 -as [int] }
        default { throw "Unknown protocol '$protocol'." }
    }


Apologies for the delay in responding.